pkg/pipeline/bean/ConfigMapBean.go

```diff
@@ -86,3 +131,42 @@ type CreateJobEnvOverridePayload struct {
 	EnvId  int   `json:"envId"`
 	UserId int32 `json:"-"`
 }
+
+func GetDecodedData(data string) (string, error) {
+	secretsList := SecretsList{}
+	err := json.Unmarshal([]byte(data), &secretsList)
+	if err != nil {
+		return "", err
+	}
+
+	for _, configData := range secretsList.ConfigData {
+		configData.Data = configData.GetDecodedData()
+	}
+
+	marshal, err := json.Marshal(secretsList)
+	if err != nil {
+		return "", err
+	}
+	return string(marshal), nil
+}
+
+func GetEncodedData(data string) (string, error) {
```

Comment:
you have written similar methods multiple times, make all of them generic

Reply:
done